### DISCUSSION TITLE
How to send action from Child to Parent?

### DISCUSSION THREAD

**Discussion by jerryga** [pointfreeco/swift-composable-architecture#2429] [Q&A] [answered]
Hello, I want to update the data of Parent State when I tap a button on the child view. And the child view has its own State, Action and Reducer. How could I do it?  Thanks in advance.

**Comment by maksimn** [comment, marked as answer]
Hello, @jerryga 
You do not need to send an action from child to parent. Child state and child action are parts of parent state and parent action by definition:

```
struct Parent: Reducer {
  struct State: Equatable {
    var something: String
    var child: Child.State
  }
  enum Action: Equatable {
    case someAction
    case child(Child.Action)
  }
}
```

And these Parent and Child are tied in the common `Store`. So, the child actions can be handled in the `Parent` reducer the same way as its own actions:

```
struct Parent: Reducer {
  // ...

  func reduce(into state: inout State, action: Action) -> Effect<Action> {
    switch action {
      // ...
      case .child(let childAction):
        // Here you can handle child actions.
   }
  }
}
```

**Reply by jerryga** [reply]
Thank you very much! You saved my time.

**Reply by jerryga** [reply]
Hello maksimn, do you know how to send an action from parent to child? I just want to call child's Action from Parent.
